src/service/cia301/co_ssdo.c

```diff
@@ -618,7 +642,12 @@ CO_ERR COSdoEndDownloadBlock(CO_SDO *srv)
         result = COObjWrBufCont(srv->Obj, srv->Node, srv->Buf.Start, len);
         if (result != CO_ERR_NONE) {
             srv->Node->Error = CO_ERR_SDO_WRITE;
-            COSdoAbort(srv, CO_SDO_ERR_TOS);
+            if (srv->Abort > 0) {
+                COSdoAbort(srv, srv->Abort);
+            } else {
+                COSdoAbort(srv, CO_SDO_ERR_TOS);
+            }
+            result = CO_ERR_SDO_ABORT;
```

Comment:
Why do we want to change the error here from whatever it is entering this if statement to the abort error? Is that because the if else you added above this always calls the abort function?